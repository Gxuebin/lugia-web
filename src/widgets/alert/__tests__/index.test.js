/**
 *
 * create by guorg
 *
 * @flow
 *
 */
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import AlertDemo from '../demo';

describe('Alert', () => {
  it('css', () => {
    const Target = <AlertDemo />;
    expect(renderer.create(Target).toJSON()).toMatchSnapshot();
  });
});
