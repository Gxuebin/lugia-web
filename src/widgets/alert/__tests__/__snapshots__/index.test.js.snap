// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert css 1`] = `
.c8 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
}

.c1 {
  position: relative;
  box-sizing: border-box;
  font-size: 1.4rem;
  overflow: hidden;
  padding: 1.7142857142857144em 0.7142857142857143em;
  line-height: 1.5;
  border-radius: 0.28571428571428575em;
  border-left: 0.28571428571428575em solid #684fff;
  background: rgba(104,79,255,0.2);
}

.c4 {
  position: relative;
  box-sizing: border-box;
  font-size: 1.4rem;
  overflow: hidden;
  padding: 1.7142857142857144em 0.7142857142857143em;
  line-height: 1.5;
  border-radius: 0.28571428571428575em;
  border-left: 0.28571428571428575em solid #56c22d;
  background: rgba(86,194,45,0.2);
}

.c5 {
  position: relative;
  box-sizing: border-box;
  font-size: 1.4rem;
  overflow: hidden;
  padding: 1.7142857142857144em 0.7142857142857143em;
  line-height: 1.5;
  border-radius: 0.28571428571428575em;
  border-left: 0.28571428571428575em solid #f22735;
  background: rgba(242,39,53,0.2);
}

.c6 {
  position: relative;
  box-sizing: border-box;
  font-size: 1.4rem;
  overflow: hidden;
  padding: 1.7142857142857144em 0.7142857142857143em;
  line-height: 1.5;
  border-radius: 0.28571428571428575em;
  border-left: 0.28571428571428575em solid #f8ac30;
  background: rgba(248,172,48,0.2);
}

.c11 {
  position: relative;
  box-sizing: border-box;
  font-size: 1.4rem;
  overflow: hidden;
  padding: 1.7142857142857144em 0.7142857142857143em;
  line-height: 1.5;
  border-radius: 0.28571428571428575em;
  background: rgba(104,79,255,0.2);
}

.c13 {
  position: relative;
  box-sizing: border-box;
  font-size: 1.4rem;
  overflow: hidden;
  padding: 1.7142857142857144em 0.7142857142857143em;
  line-height: 1.5;
  border-radius: 0.28571428571428575em;
  background: rgba(86,194,45,0.2);
}

.c14 {
  position: relative;
  box-sizing: border-box;
  font-size: 1.4rem;
  overflow: hidden;
  padding: 1.7142857142857144em 0.7142857142857143em;
  line-height: 1.5;
  border-radius: 0.28571428571428575em;
  background: rgba(242,39,53,0.2);
}

.c15 {
  position: relative;
  box-sizing: border-box;
  font-size: 1.4rem;
  overflow: hidden;
  padding: 1.7142857142857144em 0.7142857142857143em;
  line-height: 1.5;
  border-radius: 0.28571428571428575em;
  background: rgba(248,172,48,0.2);
}

.c17 {
  position: relative;
  box-sizing: border-box;
  font-size: 1.4rem;
  overflow: hidden;
  padding: 1.7142857142857144em 0.7142857142857143em;
  line-height: 1.5;
  border-radius: 0.28571428571428575em;
  border-left: 0.28571428571428575em solid #ffa500;
  background: rgba(255,165,0,0.2);
  width: 14.285714285714286em;
}

.c18 {
  position: relative;
  box-sizing: border-box;
  font-size: 1.4rem;
  overflow: hidden;
  padding: 1.7142857142857144em 0.7142857142857143em;
  line-height: 1.5;
  border-radius: 0.28571428571428575em;
  background: rgba(255,165,0,0.2);
  width: 14.285714285714286em;
}

.c2 {
  color: #333333;
  vertical-align: text-bottom;
  font-size: 1em;
}

.c16 {
  color: #333333;
  vertical-align: text-bottom;
  font-size: 1.2857142857142858em;
}

.c7 {
  overflow: hidden;
  position: absolute;
  top: 1.7142857142857144em;
  right: 1em;
}

.c9 {
  overflow: hidden;
  position: absolute;
  top: 1.7142857142857144em;
  right: 1em;
  color: #f22735;
}

.c10 {
  overflow: hidden;
  position: absolute;
  top: 1.7142857142857144em;
  right: 1em;
  color: #f8ac30;
}

.c19 {
  overflow: hidden;
  position: absolute;
  top: 1.7142857142857144em;
  right: 1em;
  color: #ffa500;
}

.c3 {
  display: block;
  color: #666666;
  font-size: 1em;
  margin-left: 0;
}

.c12 {
  display: block;
  color: #666666;
  font-size: 1em;
  margin-left: 2.142857142857143em;
}

.c0 {
  margin: 50px;
}

.c0 > div {
  margin-bottom: 16px;
}

<div
  className="c0"
>
  <div
    className="c1"
    height={null}
    type="info"
  >
    <span
      className="c2"
    >
      Alert-info
    </span>
    <span
      className="c3"
    />
  </div>
  <div
    className="c4"
    height={null}
    type="success"
  >
    <span
      className="c2"
    >
      Alert-success
    </span>
    <span
      className="c3"
    />
  </div>
  <div
    className="c5"
    height={null}
    type="error"
  >
    <span
      className="c2"
    >
      Alert-error
    </span>
    <span
      className="c3"
    />
  </div>
  <div
    className="c6"
    height={null}
    type="warning"
  >
    <span
      className="c2"
    >
      Alert-warning
    </span>
    <span
      className="c3"
    />
  </div>
  <div
    className="c1"
    height={null}
    type="info"
  >
    <span
      className="c2"
    >
      Alert-info
    </span>
    <span
      className="c3"
    />
    <a
      className="c7"
      onClick={[Function]}
      type="info"
    >
      <i
        className="lugia-icon-reminder_close  c8"
      />
    </a>
  </div>
  <div
    className="c4"
    height={null}
    type="success"
  >
    <span
      className="c2"
    >
      Alert-success
    </span>
    <span
      className="c3"
    />
    <a
      className="c7"
      onClick={[Function]}
      type="success"
    >
      <i
        className="lugia-icon-reminder_close  c8"
      />
    </a>
  </div>
  <div
    className="c5"
    height={null}
    type="error"
  >
    <span
      className="c2"
    >
      Alert-error
    </span>
    <span
      className="c3"
    />
    <a
      className="c9"
      onClick={[Function]}
      type="error"
    >
      确定
    </a>
  </div>
  <div
    className="c6"
    height={null}
    type="warning"
  >
    <span
      className="c2"
    >
      Alert-warning
    </span>
    <span
      className="c3"
    />
    <a
      className="c10"
      onClick={[Function]}
      type="warning"
    >
      ok
    </a>
  </div>
  <div
    className="c11"
    height={null}
    type="info"
  >
    <i
      className="lugia-icon-reminder_info_circle  c8"
    />
    <span
      className="c2"
    >
      Alert-info
    </span>
    <span
      className="c12"
    />
  </div>
  <div
    className="c13"
    height={null}
    type="success"
  >
    <i
      className="lugia-icon-reminder_check_circle  c8"
    />
    <span
      className="c2"
    >
      Alert-success
    </span>
    <span
      className="c12"
    />
  </div>
  <div
    className="c14"
    height={null}
    type="error"
  >
    <i
      className="lugia-icon-reminder_close_circle  c8"
    />
    <span
      className="c2"
    >
      Alert-error
    </span>
    <span
      className="c12"
    />
  </div>
  <div
    className="c15"
    height={null}
    type="warning"
  >
    <i
      className="lugia-icon-reminder_exclamation_circle  c8"
    />
    <span
      className="c2"
    >
      Alert-warning
    </span>
    <span
      className="c12"
    />
  </div>
  <div
    className="c1"
    height={null}
    type="info"
  >
    <span
      className="c16"
    >
      Alert-info
    </span>
    <span
      className="c3"
    >
      这是辅助性文字介绍
    </span>
    <a
      className="c7"
      onClick={[Function]}
      type="info"
    >
      <i
        className="lugia-icon-reminder_close  c8"
      />
    </a>
  </div>
  <div
    className="c4"
    height={null}
    type="success"
  >
    <span
      className="c16"
    >
      Alert-success
    </span>
    <span
      className="c3"
    >
      这是辅助性文字介绍
    </span>
    <a
      className="c7"
      onClick={[Function]}
      type="success"
    >
      <i
        className="lugia-icon-reminder_close  c8"
      />
    </a>
  </div>
  <div
    className="c5"
    height={null}
    type="error"
  >
    <span
      className="c16"
    >
      Alert-error
    </span>
    <span
      className="c3"
    >
      这是辅助性文字介绍
    </span>
    <a
      className="c9"
      onClick={[Function]}
      type="error"
    >
      确定
    </a>
  </div>
  <div
    className="c6"
    height={null}
    type="warning"
  >
    <span
      className="c16"
    >
      Alert-warning
    </span>
    <span
      className="c3"
    >
      这是辅助性文字介绍
    </span>
    <a
      className="c10"
      onClick={[Function]}
      type="warning"
    >
      ok
    </a>
  </div>
  <div
    className="c11"
    height={null}
    type="info"
  >
    <i
      className="lugia-icon-reminder_info_circle  c8"
    />
    <span
      className="c16"
    >
      Alert-info
    </span>
    <span
      className="c12"
    >
      这是辅助性文字介绍
    </span>
    <a
      className="c7"
      onClick={[Function]}
      type="info"
    >
      <i
        className="lugia-icon-reminder_close  c8"
      />
    </a>
  </div>
  <div
    className="c13"
    height={null}
    type="success"
  >
    <i
      className="lugia-icon-reminder_check_circle  c8"
    />
    <span
      className="c16"
    >
      Alert-success
    </span>
    <span
      className="c12"
    >
      这是辅助性文字介绍
    </span>
    <a
      className="c7"
      onClick={[Function]}
      type="success"
    >
      <i
        className="lugia-icon-reminder_close  c8"
      />
    </a>
  </div>
  <div
    className="c14"
    height={null}
    type="error"
  >
    <i
      className="lugia-icon-reminder_close_circle  c8"
    />
    <span
      className="c16"
    >
      Alert-error
    </span>
    <span
      className="c12"
    >
      这是辅助性文字介绍
    </span>
    <a
      className="c9"
      onClick={[Function]}
      type="error"
    >
      确定
    </a>
  </div>
  <div
    className="c15"
    height={null}
    type="warning"
  >
    <i
      className="lugia-icon-reminder_exclamation_circle  c8"
    />
    <span
      className="c16"
    >
      Alert-warning
    </span>
    <span
      className="c12"
    >
      这是辅助性文字介绍
    </span>
    <a
      className="c10"
      onClick={[Function]}
      type="warning"
    >
      ok
    </a>
  </div>
  <div
    className="c11"
    height={null}
    type="info"
  >
    <i
      className="lugia-icon-logo_twitter  c8"
    />
    <span
      className="c2"
    >
      Alert-info
    </span>
    <span
      className="c12"
    />
  </div>
  <p>
    Theme
  </p>
  <div
    className="c17"
    height={null}
    type="info"
  >
    <span
      className="c2"
    >
      Alert-info
    </span>
    <span
      className="c3"
    />
  </div>
  <div
    className="c18"
    height={null}
    type="info"
  >
    <i
      className="lugia-icon-reminder_info_circle  c8"
    />
    <span
      className="c2"
    >
      Alert-info
    </span>
    <span
      className="c12"
    />
  </div>
  <div
    className="c18"
    height={null}
    type="info"
  >
    <i
      className="lugia-icon-reminder_info_circle  c8"
    />
    <span
      className="c2"
    >
      Alert-info
    </span>
    <span
      className="c12"
    />
    <a
      className="c19"
      onClick={[Function]}
      type="info"
    >
      ok
    </a>
  </div>
</div>
`;
